README.md

```diff
@@ -20,48 +20,187 @@ const { Reshuffle } = require('reshuffle')
+
+const family = await mysql.query(
+  "SELECT firstName, lastName, age FROM users WHERE lastName='Coltrane'"
+)
+
+const avgResponse = await mysql.query(
+  "SELECT average(age) AS avg FROM users WHERE lastName='Coltrane'"
+)
+const averageAge = avgResponse.rows[0].avg
+
+```
+
+The `query` action can be used to run _any_ SQL command on the connected
+database (not just `SELECT`). The query is defined in the `sql` string. The
+optional `params` can be used to generate parameterized queries, as shown in
+the following example:
+
+```js
+const age = await mysql.query(
+  "SELECT age FROM users WHERE firstName = ? and lastName = ?",
+  ['John', 'Smith']
+)
 ```
 
-More details and code samples about Node MySQL 2 Client can be found [here](https://www.npmjs.com/package/mysql2)
+This action returns an object with the results of the query, where
+`fields` is an array of all field names, as returned by the query.
+Field names in a `SELECT` query are column names, or are specified
+with an `AS` clause. Every element of `rows` is uses the names in
+`fields` as its object keys.
 
+Note that every call to `query` may use a different database connection.
+You can use the [sequence](#sequence) or [transaction](#transaction) actions
+if such a guarantee is required.
 
-##### <a name="sdkPromise"></a>Full access to the MySQL Client SDK with support of Promise API
+##### <a name="sequence"></a>Sequence action
 
 
 ```js
-const sdkPromise = mysql.sdkPromise()
+const res = await mysql.sequence(async (query) => {
+  await query("INSERT INTO users VALUES ('Miles', 'Davis', 43)")
+  return query("SELECT COUNT(*) FROM users where 1")
+})
+const userCount = res.rows[0].count
+
 ```
 
+Use `sequence` to perform multiple queries on the same database connection.
+This action receives a `seq` function that may issue queries to the database,
+all of which are guaranteed to run through the same connection. `seq` gets
+one argument, which is a `query` function that can be used the same way as
+the [query](#query) action. `seq` may, of course, use any JavaScript code to
+implement its logic, log to the console etc.
+
+Note that while `sequence` uses the same connection to run all queries, it
+does not offer the transactional guarantees offered by
+[transaction](#transaction). You can use it for weak isolation models, or
+construct transactions directly without using `transaction`.
+
+##### <a name="transaction"></a>Transaction action
+
+
+```js
+await mysql.transaction(async (query) => {
+  const res = await query("SELECT COUNT(*) FROM users")
+  const userCount = res.rows[0].count
+  if (100 <= userCount) {
+    throw new Error('Too many users:', userCount)
+  }
+  return query("INSERT INTO users VALUES ('Charlie', 'Parker', 49)")
+})
+```
+
+Use `transaction` to run multiple queries as an atomic SQL transaction.
+The interface is identical to the [sequence action](#sequence), but all
+the queries issued `seq` either success or fail together. If any of the
```

Comment:
I think the `seq` here is redundant?

Reply:
It's an explanation about the differences between the transaction and sequence actions. The same description is in pgsql connector.